test/resources/Hooks/useCopied/index.test.js

```diff
@@ -0,0 +1,53 @@
+/** @jest-environment jsdom */
+import { renderHook, act } from '@testing-library/react'
+import useCopiedMessage from '../../../../resources/Hooks/useCopiedMessage'
+import link from '../../../../resources/link'
+
+jest.mock('../../../../resources/link', () => ({
+  send: jest.fn()
+}))
+
+describe('#useCopied', () => {
+  jest.useFakeTimers()
+
+  let result
+
+  const getBool = () => result.current[0]
+  const copyFn = () => result.current[1]()
+
+  beforeEach(() => {
+    ;({ result } = renderHook(() => useCopiedMessage('VALUE')))
+    jest.clearAllMocks()
+  })
+
+  it('should correctly initialise a boolean to check if the copiedMessage should be shown', () => {
```

Comment:
the test name should describe the desired behavior more than talking about the implementation or internals. so something like "should not show the copied message by default"

also the test can be written to make the usage of the hook a little more obvious, using it how it would be used in an actual component. as of now it's a little hard to tell what `result.current[0]` is and why it should be false. you could do something like this:

above:
```
const render = () => renderHook(() => useCopiedMessage('VALUE')).result.current
```
and in the test:
```
const [showCopiedMessage, copyMessage] = render()
  
act(() => {
  copyMessage()
})

expect(showCopiedMessage).toBe(true)
```

Reply:
circling back on this and I guess it doesn't work this way, which is a shame. it seems like a commonly accepted practice is to create a test component so you can use it the way its used in the code, but I don't love that either as it introduces more ways to make a mistake into the process. I'd say lets stick w. what you have except run the test and assign the result in every test so its a little more obvious where those values are coming from.